intra_process_comms/src/intrapc.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <functional>
+#include <iostream>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <std_msgs/msg/string.hpp>
+
+void on_message(const std_msgs::msg::String::ConstSharedPtr & msg)
+{
+  std::cout << "I heard [" << msg->data << "]" << std::endl;
+  rclcpp::sleep_for(1_s);
+  if (msg->data.at(msg->data.length() - 1) == '9')
+  {
+    raise(SIGINT);
+  }
+}
+
+void on_timer(rclcpp::Publisher::SharedPtr & publisher, int & i)
+{
+  std_msgs::msg::String::SharedPtr msg(new std_msgs::msg::String());
+  msg->data = "Hello World: " + std::to_string(i++);
+  if (publisher) {
+    std::cout << "Publishing: '" << msg->data << "'" << std::endl;
+    publisher->publish(msg);
+  } else {
+    std::cerr << "Invalid publisher!" << std::endl;
+  }
+}
+
+int main(int argc, char * argv[])
+{
+  rclcpp::init(argc, argv);
+
+  auto node1 = rclcpp::Node::make_shared("node1");
+  auto node2 = rclcpp::Node::make_shared("node2");
```

Comment:
Don't they have to enable intra process comm here (like https://github.com/ros2/system_tests/pull/28/files#diff-fe711788916f61638c377d72e77584a5R28)?